packages/devextreme/js/__internal/scheduler/appointment_popup/appointment_popup.test.ts

```diff
@@ -0,0 +1,952 @@
+  });
+
+  if (!target) throw new Error(`Radio with value "${value}" not found`);
+
+  radios.forEach((r) => {
+    r.setAttribute('aria-checked', 'false');
+    r.classList.remove('dx-item-selected', 'dx-radiobutton-checked');
+  });
+
+  target.setAttribute('aria-checked', 'true');
+  target.classList.add('dx-item-selected', 'dx-radiobutton-checked');
+
+  target.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+
+  return target;
+}
+
+function getSelectedRadio(container: HTMLElement): HTMLElement | null {
+  return container.querySelector('[role="radio"][aria-checked="true"]');
+}
+
+function getSelectedRadioValue(container: HTMLElement): string | null {
+  const selected = getSelectedRadio(container);
+  return selected?.getAttribute('aria-label') ?? selected?.textContent?.trim() ?? null;
+}
+
+describe('Appointment popup form', () => {
+  beforeEach(() => {
+    fx.off = true;
+    setupSchedulerTestEnvironment();
+  });
+
+  afterEach(() => {
+    fx.off = false;
+    document.body.innerHTML = '';
+    jest.useRealTimers();
+  });
+
+  it('Original appointment\'s fields shouldn\'t fill if used fieldExpr', async () => {
+    const data: Record<string, unknown>[] = [];
+    const textExpValue = 'Subject';
+
+    let newAppointment: Record<string, unknown> | null = null;
+
+    const { scheduler, POM } = await createScheduler({
+      dataSource: data,
+      views: ['week'],
+      currentView: 'week',
+      currentDate: new Date(2021, 4, 27),
+      textExpr: textExpValue,
+      onAppointmentAdded: ({ appointmentData }) => {
+        newAppointment = appointmentData;
+      },
+      height: 600,
+    });
+
+    scheduler.showAppointmentPopup();
+
+    const form = POM.getForm() as HTMLElement;
+    setInputValueByLabel('Subject', 'qwerty', form);
+
+    POM.clickDoneButton();
+
+    expect(newAppointment?.[textExpValue]).toBe('qwerty');
+    // @ts-expect-error set the value in the promise
+    expect(newAppointment?.text).toBeUndefined();
+
+    expect(data[0].Subject).toBe('qwerty');
+    expect(data[0].text).toBeUndefined();
+  });
+
+  it('Recurrence form should work properly if recurrenceRule property mapped recurrenceRuleExpr', async () => {
+    const { scheduler, POM } = await createScheduler({
+      dataSource: [{
+        text: 'Watercolor Landscape',
+        startDate: new Date(2017, 4, 1, 9, 30),
+        endDate: new Date(2017, 4, 1, 11),
+        customRecurrenceRule: 'FREQ=WEEKLY;BYDAY=TU,FR;COUNT=10',
+      }],
+      views: ['month'],
+      currentView: 'month',
+      currentDate: new Date(2017, 4, 25),
+      recurrenceRuleExpr: 'customRecurrenceRule',
+      height: 600,
+    });
+
+    openPopupByDblClick(POM);
+
+    const editSeriesButton = document.querySelector('[aria-label="Edit series"]') as HTMLElement;
+    editSeriesButton.click();
+
+    const { value: repeatInputRecurrence } = getSwitchByName('repeat');
+
+    scheduler.hideAppointmentPopup();
+    scheduler.showAppointmentPopup();
```

Comment:
I don't understand why an what do you check:
```
    scheduler.hideAppointmentPopup();
    scheduler.showAppointmentPopup();
```
and why expect of `repeatInputRecurrence` is at the end of the test. I would like read test line by line without jumping up and down.

Maybe you have to rename test. It's unclear what is going on here. show, hide, show, hide, expect.

Reply:
Okey, I see, it's old tests. It's require to keep git history for these tests. Are you sure you want to move it to the jest and not just keep in QUnits?
So how do we do this:
commit 1: move QUnit file into jest directory
commit 2: replace all QUnit tests code into jest

Then ask Devops team to help merge it with rebase. It will save the git history.

By the way did you use AI?